back/engines/commercial/id_clave_unica/spec/requests/clave_unica_verification_spec.rb

```diff
@@ -224,20 +224,57 @@ def expect_to_create_verified_and_identified_user(user)
         user = User.order(created_at: :asc).last
         expect_to_create_verified_and_identified_user(user)
 
-        token = AuthToken::AuthToken.new(payload: user.to_token_payload).token
-        headers = { 'Authorization' => "Bearer #{token}" }
+        headers = { 'Authorization' => authorization_header(user) }
+
         post '/web_api/v1/user/resend_code', params: { new_email: 'newcoolemail@example.org' }, headers: headers
         expect(response).to have_http_status(:ok)
         expect(user.reload).to have_attributes({ new_email: 'newcoolemail@example.org' })
         expect(user.confirmation_required?).to be(true)
 
         post '/web_api/v1/user/confirm', params: { confirmation: { code: user.email_confirmation_code } }, headers: headers
+
         expect(response).to have_http_status(:ok)
         expect(user.reload.confirmation_required?).to be(false)
         expect(user).to have_attributes({ email: 'newcoolemail@example.org' })
         expect(user.new_email).to be_nil
       end
 
+      it 'creates user that cannot update her email' do
+        get '/auth/clave_unica'
+        follow_redirect!
+
+        user = User.order(created_at: :asc).last
+
+        headers = { 'Authorization' => authorization_header(user) }
+        patch "/web_api/v1/users/#{user.id}", params: { user: { email: 'newcoolemail@example.org' } }, headers: headers
+
+        expect(response).not_to have_http_status(:ok)
+        expect(user.reload).to have_attributes({ email: nil })
+        expect(user.confirmation_required?).to be(true)
+      end
+
+      it 'creates user that can enter email, change it and then confirm it' do
+        get '/auth/clave_unica'
+        follow_redirect!
+
+        user = User.order(created_at: :asc).last
+        headers = { 'Authorization' => authorization_header(user) }
+
+        post '/web_api/v1/user/resend_code', params: { new_email: '1@example.org' }, headers: headers
+        expect(response).to have_http_status(:ok)
+        expect(user.reload).to have_attributes({ new_email: '1@example.org' })
+
+        post '/web_api/v1/user/resend_code', params: { new_email: '2@example.org' }, headers: headers
+        expect(response).to have_http_status(:ok)
+        expect(user.reload).to have_attributes({ new_email: '2@example.org' })
+
+        post '/web_api/v1/user/confirm', params: { confirmation: { code: user.email_confirmation_code } }, headers: headers
+        expect(response).to have_http_status(:ok)
+        expect(user.reload.confirmation_required?).to be(false)
+        expect(user).to have_attributes({ email: '2@example.org' })
+        expect(user.new_email).to be_nil
+      end
+
```

Comment:
Be great to have more of these multi-stage tests for different user scenarios such as returning lite users, user changing their email and needing to reconfirm etc

Reply:
I agree. Added [a ticket for it](https://www.notion.so/citizenlab/Add-more-multi-stage-acceptance-test-for-different-auth-scenarious-4ebb4f22accb4fbba44868459aeb6786). Let's see if we have time for it.